satpy/tests/reader_tests/test_atms_sdr_hdf5.py

```diff
@@ -0,0 +1,337 @@
+            file_content[k + "/shape"] = new_shape
+
+    @staticmethod
+    def _add_geo_ref(file_content, filename):
+        geo_prefix = 'GATMO'
+        file_content['/attr/N_GEO_Ref'] = geo_prefix + filename[5:]
+
+    @staticmethod
+    def _convert_numpy_content_to_dataarray(final_content):
+        import dask.array as da
+        from xarray import DataArray
+        for key, val in final_content.items():
+            if isinstance(val, np.ndarray):
+                val = da.from_array(val, chunks=val.shape)
+                if val.ndim > 2:
+                    final_content[key] = DataArray(val, dims=('y', 'x', 'z'))
+                elif val.ndim > 1:
+                    final_content[key] = DataArray(val, dims=('y', 'x'))
+                else:
+                    final_content[key] = DataArray(val)
+
+    def get_test_content(self, filename, filename_info, filetype_info):
+        """Mimic reader input file content."""
+        final_content = {}
+        for dataset in self.datasets:
+            dataset_group = DATASET_KEYS[dataset]
+            prefix1 = 'Data_Products/{dataset_group}'.format(dataset_group=dataset_group)
+            prefix2 = '{prefix}/{dataset_group}_Aggr'.format(prefix=prefix1, dataset_group=dataset_group)
+            prefix3 = 'All_Data/{dataset_group}_All'.format(dataset_group=dataset_group)
+
+            file_content = {}
+            self._add_basic_metadata_to_file_content(file_content, filename_info, self._num_test_granules)
+            self._add_granule_specific_info_to_file_content(file_content, dataset_group,
+                                                            self._num_test_granules, self._num_scans_per_gran,
+                                                            prefix1)
+            self._add_geo_ref(file_content, filename)
+
+            for k, v in list(file_content.items()):
+                file_content[k.format(prefix1=prefix1, prefix2=prefix2, prefix3=prefix3)] = v
+
+            if filename[:5] in ['SATMS', 'TATMS']:
+                self._add_data_info_to_file_content(file_content, filename, prefix3,
+                                                    self._num_test_granules)
+            elif filename[0] == 'G':
+                self._add_geolocation_info_to_file_content(file_content, filename, prefix3,
+                                                           self._num_test_granules)
+            final_content.update(file_content)
+        self._convert_numpy_content_to_dataarray(final_content)
+        return final_content
+
+    def _adjust_scaling_factors(self, factors, file_units, output_units):
+        """Adjust scaling factors ."""
+        if file_units == output_units:
+            return factors
+        else:
+            factors = np.array([1.0, 0], dtype=np.float32)
+            factors = xr.DataArray(da.from_array(factors, chunks=1))
+        return factors
+
+
+class TestATMS_SDR_Reader(unittest.TestCase):
```

Comment:
We try to move away from unittest and use pytest-style tests nowadays. Can this class be updated to follow this?

Reply:
Well yes, I agree. But I am very much building upon how it was already done for VIIRS. So this is no consistent with how the viirs-sdr tests are formulated. I am either not completely comfortable with how the mock part works in the `setUp`, so it it feels not trivial to me how to migrate it to using pytest and fixtures. I would rather leave it for another refactoring PR, and then doing both VIIRS and ATMS together.

Reply:
I did as suggested now, thought it required more work, was not acquainted with the `setup_method`/`teardown_method` methods. Thanks!